### PR TITLE
WIP: Tanstack transactions

### PR DESCRIPTION
**Pull request by nop33**

### CODE REVIEW

File: apps/desktop-wallet/src/api/addresses.ts [hunk 1]
```diff
@@ -17,63 +17,15 @@ along with the library. If not, see <http://www.gnu.org/licenses/>.
...
+import { TOTAL_NUMBER_OF_GROUPS } from '@alephium/web3'
 
-import { Address, AddressTransactionsSyncResult } from '@/types/addresses'
 import {
   deriveAddressesInGroup,
   getGapFromLastActiveAddress,
   splitResultsArrayIntoOneArrayPerGroup
 } from '@/utils/addresses'
 
-export const fetchAddressesTransactions = async (
```
Comment (nop33) [inline apps/desktop-wallet/src/api/addresses.ts:30]: And goodbye last remaining Redux async actions

File: apps/desktop-wallet/src/api/apiDataHooks/address/useFetchAddressBalancesAlph.ts [hunk 2]
```diff
@@ -30,22 +30,21 @@ interface UseAddressAlphBalancesProps extends SkipProp {
 
 const useFetchAddressBalancesAlph = ({ addressHash, skip }: UseAddressAlphBalancesProps) => {
   const networkId = useAppSelector((s) => s.network.settings.networkId)
-  const queryProps = { addressHash, networkId }
-
-  const { data: txHashes, isLoading: isLoadingTxHashes } = useQuery(addressLatestTransactionHashQuery(queryProps))
 
+  const { data: updatesSignal, isLoading: isLoadingUpdatesSignal } = useFetchAddressUpdatesSignal({ addressHash, skip })
```
Comment (nop33) [inline apps/desktop-wallet/src/api/apiDataHooks/address/useFetchAddressBalancesAlph.ts:34]: I decided to rename `useFetchAddressLatestTransaction` to something that is more accurate based on its usage. It's used as a "signal" to whether all other queries should invalidate or not.

File: apps/desktop-wallet/src/api/apiDataHooks/address/useFetchAddressInfiniteTransactions.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  const networkId = useAppSelector((s) => s.network.settings.networkId)
+
+  const [fetchedTransactionListAt, setRefreshedTransactionListAt] = useState(0)
+  const refresh = useCallback(() => setRefreshedTransactionListAt(new Date().getTime()), [])
+
+  const { data: updatesSignal, isLoading: isLoadingUpdatesSignal } = useFetchAddressUpdatesSignal({ addressHash })
+  const { data, fetchNextPage, isLoading, hasNextPage, isFetchingNextPage } = useInfiniteQuery(
+    addressTransactionsInfiniteQuery({
+      addressHash,
+      timestamp: fetchedTransactionListAt,
```
Comment (nop33) [inline apps/desktop-wallet/src/api/apiDataHooks/address/useFetchAddressInfiniteTransactions.ts:41]: When the user clicks the "Refresh" row, a new timestamp is being used as a `queryKey`, forcing the whole query to re-fetch.

File: apps/desktop-wallet/src/api/apiDataHooks/address/useFetchAddressInfiniteTransactions.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      networkId,
+      skip: isLoadingUpdatesSignal
+    })
+  )
+
+  const fetchedConfirmedTxs = useMemo(() => data?.pages.flat() ?? [], [data?.pages])
+  const latestFetchedTxHash = fetchedConfirmedTxs[0]?.hash
+  const latestUnfetchedTxHash = updatesSignal?.latestTx?.hash
+  const showNewTxsMessage = !isLoading && latestUnfetchedTxHash && latestFetchedTxHash !== latestUnfetchedTxHash
```
Comment (nop33) [inline apps/desktop-wallet/src/api/apiDataHooks/address/useFetchAddressInfiniteTransactions.ts:50]: We are showing the "Refresh" row when we detect that the latest hash that we haven't yet fetched in the tx list is not the same as the latest one from the already fetched ones.

File: apps/desktop-wallet/src/api/apiDataHooks/transaction/useFetchTransaction.tsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+import { useAppSelector } from '@/hooks/redux'
+
+interface UseFetchTransactionProps {
+  txHash: string
+}
+
+const useFetchTransaction = ({ txHash }: UseFetchTransactionProps) => {
+  const sentTx = useAppSelector((s) => selectSentTransactionByHash(s, txHash))
+
+  const isPendingTx = sentTx && sentTx.status !== 'confirmed'
```
Comment (nop33) [inline apps/desktop-wallet/src/api/apiDataHooks/transaction/useFetchTransaction.tsx:32]: See later for how sent transactions are structured in the slice

File: apps/desktop-wallet/src/api/apiDataHooks/transaction/useFetchTransaction.tsx [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+  const { data: confirmedTx, isLoading: isLoadingConfirmedTx } = useQuery(
+    confirmedTransactionQuery({ txHash, skip: isPendingTx })
+  )
+  const { data: pendingTx, isLoading: isLoadingPendingTx } = useQuery(
+    pendingTransactionQuery({ txHash, skip: !isPendingTx })
+  )
```
Comment (nop33) [inline apps/desktop-wallet/src/api/apiDataHooks/transaction/useFetchTransaction.tsx:39]: Both of these are using the same endpoint, but with different tanstack config. The first one as `staleTime: Infinity` since its data will never change, and the second one has `refetchInterval: 3000`

File: apps/desktop-wallet/src/components/SnackbarManager.tsx [hunk 7]
```diff
@@ -16,22 +16,43 @@ You should have received a copy of the GNU Lesser General Public License
...
+  const sentTxs = useAppSelector(selectAllSentTransactions)
 
-  const message = messages.length > 0 ? messages[0] : undefined
+  return (
+    <ModalPortal>
+      {(messages.length > 0 || sentTxs.length > 0) && (
+        <SnackbarManagerContainer>
+          {sentTxs.map((sentTxs) => (
+            <SentTransactionSnackbarPopup key={sentTxs.hash} txHash={sentTxs.hash} />
+          ))}
```
Comment (nop33) [inline apps/desktop-wallet/src/components/SnackbarManager.tsx:42]: The send transactions boxes will render together with any other snackbar

File: apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+  const sentTx = useAppSelector((s) => selectSentTransactionByHash(s, txHash))
+  const [hide, setHide] = useState(false)
+
+  const { data: tx } = useQuery(pendingTransactionQuery({ txHash, skip: !sentTx || sentTx.status === 'confirmed' }))
+
+  useEffect(() => {
+    if (!tx) return
+
+    dispatch(sentTransactionStatusChanged({ hash: tx.hash, status: isConfirmedTx(tx) ? 'confirmed' : 'mempooled' }))
+  }, [dispatch, tx])
```
Comment (nop33) [inline apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx:51]: Updating the state in Redux based on results of Tanstack query

File: apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+    let timer: ReturnType<typeof setTimeout>
+
+    if (sentTx?.status === 'confirmed') {
+      timer = setTimeout(() => setHide(true), 5000)
+    }
+
+    return () => {
+      if (timer) clearTimeout(timer)
+    }
+  }, [sentTx?.status])
```
Comment (nop33) [inline apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx:63]: I wanna keep the snackbar visible for a few seconds after the sent tx has confirmed

File: apps/desktop-wallet/src/features/transactionsDisplay/TransactionList.tsx
Comment (nop33) [inline apps/desktop-wallet/src/features/transactionsDisplay/TransactionList.tsx:1]: This file was split to 3 files, one for each tx list that we have:
- overview limited wallet tx list
- transfers wallet tx list (with infinite loader)
- address tx list (with infinite loading)

File: apps/desktop-wallet/src/features/transactionsDisplay/transactionDetailsModal/AddressesDataRows.tsx [hunk 10]
```diff
@@ -24,20 +25,24 @@ import AddressBadge from '@/components/AddressBadge'
...
+import { selectPendingSentTransactionByHash } from '@/features/sentTransactions/sentTransactionsSelectors'
+import PendingSentAddressBadge from '@/features/transactionsDisplay/transactionDetailsModal/PendingSentAddressBadge'
+import { TransactionDetailsModalTxProps } from '@/features/transactionsDisplay/transactionDetailsModal/types'
 import useOnAddressClick from '@/features/transactionsDisplay/transactionDetailsModal/useOnAddressClick'
 import useOpenTxInExplorer from '@/features/transactionsDisplay/transactionDetailsModal/useOpenTxInExplorer'
 import useTransactionDirection from '@/features/transactionsDisplay/useTransactionDirection'
+import { useAppSelector } from '@/hooks/redux'
 
-const AddressesDataRows = ({ tx, addressHash }: TransactionDetailsModalSectionProps) => {
+const AddressesDataRows = ({ tx, refAddressHash }: TransactionDetailsModalTxProps) => {
```
Comment (nop33) [inline apps/desktop-wallet/src/features/transactionsDisplay/transactionDetailsModal/AddressesDataRows.tsx:36]: I renamed `addressHash` to `refAddressHash` because in the context of a "transaction" we always want to know "which address as reference" do we want to display info with? Displaying a "moved" transaction in the context of a wallet will appear as "incoming" in the context of an address. So, `refAddressHash`.

File: apps/desktop-wallet/src/pages/UnlockedWallet/TransfersPage/index.tsx [hunk 11]
```diff
@@ -50,7 +50,7 @@ const TransfersPage = ({ className }: TransfersPageProps) => {
   const [selectedTokensIds, setSelectedTokensIds] = useState<TokenId[]>()
 
   useEffect(() => {
-    scrollDirection?.onChange(setDirection)
+    scrollDirection?.on('change', setDirection)
```
Comment (nop33) [inline apps/desktop-wallet/src/pages/UnlockedWallet/TransfersPage/index.tsx:53]: Finally fixing this annyong warning in console

File: apps/desktop-wallet/src/pages/UnlockedWallet/UnlockedWalletLayout.tsx [hunk 12]
```diff
@@ -153,24 +136,7 @@ const UnlockedWalletLayout = ({ children, title, className }: UnlockedWalletLayo
...
-              squared
-              short
-              onClick={refreshAddressesData}
-              disabled={isLoadingData}
-              aria-label={t('Refresh')}
-              data-tooltip-id="default"
-              data-tooltip-content={t('Refresh data')}
-            >
-              {isLoadingData ? <Spinner /> : <RefreshCw />}
-            </RefreshButton>
```
Comment (nop33) [inline apps/desktop-wallet/src/pages/UnlockedWallet/UnlockedWalletLayout.tsx:171]: We no longer need this refresh button, do we? Now we have polling.

File: apps/desktop-wallet/src/storage/addresses/addressesActions.ts [hunk 13]
```diff
@@ -59,83 +40,6 @@ export const addressSettingsSaved = createAction<{ addressHash: AddressHash; set
   'addresses/addressSettingsSaved'
 )
 
-export const syncAddressesData = createAsyncThunk(
```
Comment (nop33) [inline apps/desktop-wallet/src/storage/addresses/addressesActions.ts:62]: ✨  goodbye

File: apps/desktop-wallet/src/storage/addresses/addressesAdapters.ts [hunk 14]
```diff
@@ -21,11 +21,6 @@ import { createEntityAdapter } from '@reduxjs/toolkit'
...
 export const addressesAdapter = createEntityAdapter<Address>({
-  selectId: (address) => address.hash,
-  sortComparer: (a, b) => {
-    // Always keep main address to the top of the list
-    if (a.isDefault) return -1
-    if (b.isDefault) return 1
-    return (b.lastUsed ?? 0) - (a.lastUsed ?? 0)
-  }
+  selectId: (address) => address.hash
+  // TODO: Find a new way to get sorted addresses
```
Comment (nop33) [inline apps/desktop-wallet/src/storage/addresses/addressesAdapters.ts:25]: I would like to work on this in a follow up, this PR already grew more than I wanted.

File: apps/desktop-wallet/src/storage/addresses/addressesSlice.ts [hunk 15]
```diff
@@ -68,13 +52,6 @@ const addressesSlice = createSlice({
   reducers: {},
   extraReducers(builder) {
     builder
-      .addCase(syncingAddressDataStarted, (state) => {
```
Comment (nop33) [inline apps/desktop-wallet/src/storage/addresses/addressesSlice.ts:71]: such cleaner address slice, with NO api data anymore in it

File: apps/explorer/src/pages/AddressInfoPage/AddressTransactionRow.tsx [hunk 16]
```diff
@@ -54,7 +54,7 @@ const AddressTransactionRow = ({ transaction: tx, addressHash, isInContract }: A
   const { assets, infoType } = useTransactionInfo(tx, addressHash)
 
   const isMoved = infoType === 'move'
-  const isPending = isMempoolTx(tx)
+  const isPending = !isConfirmedTx(tx)
```
Comment (nop33) [inline apps/explorer/src/pages/AddressInfoPage/AddressTransactionRow.tsx:57]: the following diffs you'll notice that I am using 
`Transaction | PendingTransaction | MempoolTransaction`, even though the last 2 are basically the same with the difference that `PendingTransaction` includes a `type: string` field. Another difference is that the `AcceptedTransaction | PendingTransaction` is a return value of the `GET /transactions/{transactionHash}` endpoint whereas `MempoolTransaction[]` is the return value of the mempool endpoint.

It'd be nice if we could clean these up, but until then this is my proposed workaround.

File: apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx [hunk 17]
```diff
@@ -0,0 +1,183 @@
...
+      <Columns>
+        <Rows>
+          <Message status={sentTx.status} />
+
+          <HashEllipsedStyled hash={txHash} tooltipText={t('Copy hash')} showSnackbarOnCopied={false} />
+
+          <Progress status={sentTx.status} />
+        </Rows>
+        <ButtonsRow>
+          {sentTx.status !== 'sent' && <DetailsButton hash={txHash} />}
```
Comment (mvaivre) [inline apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx:78]: Note: I'm going to redesign this a bit, but the idea to open the TX modal is great!

File: apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx [hunk 18]
```diff
@@ -0,0 +1,183 @@
...
+    if (status === 'sent') {
+      setProgress(0.1)
+    } else if (status === 'mempooled') {
+      setProgress((prevValue) => (prevValue < 0.25 ? 0.25 : prevValue))
+    } else if (status === 'confirmed') {
+      setProgress(1)
+    }
+  }, [status])
+
+  useInterval(() => setProgress((prevValue) => prevValue + 0.015), 1000, status === 'confirmed' || progress > 0.9)
```
[the remark this answers is not in the document]
Comment (mvaivre) [reply apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx:118]: UI comment: after testing it, I think I'd rather have a spinner which transforms in a nice "check" animation when the TX is confirmed, rather than a fake progress bar.

File: apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx
Comment (mvaivre) [inline apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx:1]: Should the `sentTransactions` folder be in the `send` functionality folder? 🤔

File: apps/desktop-wallet/src/features/send/Transfer/index.tsx
Comment (mvaivre) [inline apps/desktop-wallet/src/features/send/Transfer/index.tsx:1]: Not directly linked to this PR, but I'd rather have a dedicated `TransferSendModal.tsx` file in the `send` folder.
This `Transfer` folder is confusing (even more because it uses an upper case first letter IMO).

File: apps/desktop-wallet/src/features/sentTransactions/sentTransactionsSlice.ts [hunk 19]
```diff
@@ -0,0 +1,49 @@
...
+  extraReducers(builder) {
+    builder
+      .addCase(transactionSent, sentTransactionsAdapter.addOne)
+      .addCase(receiveTestnetTokens.fulfilled, sentTransactionsAdapter.addOne)
+      .addCase(sentTransactionStatusChanged, (state, { payload: { hash, status } }) => {
+        sentTransactionsAdapter.updateOne(state, { id: hash, changes: { status } })
+      })
+      .addCase(walletLocked, () => initialState)
+      .addCase(walletSwitched, () => initialState)
+      .addCase(activeWalletDeleted, () => initialState)
```
Comment (mvaivre) [inline apps/desktop-wallet/src/features/sentTransactions/sentTransactionsSlice.ts:45]: Nit: we could use `builder.addMatcher(isAnyOf...` to DRY things off.

File: apps/desktop-wallet/src/features/transactionsDisplay/transactionDetailsModal/AddressesDataRows.tsx (hunk 10 above)
Comment (mvaivre) [reply apps/desktop-wallet/src/features/transactionsDisplay/transactionDetailsModal/AddressesDataRows.tsx:36]: Makes sense!

File: apps/desktop-wallet/src/features/transactionsDisplay/transactionLists/lists/AddressTransactionsList.tsx [hunk 20]
```diff
@@ -0,0 +1,93 @@
...
+import TransactionsListFooter from '@/features/transactionsDisplay/transactionLists/TransactionsListFooter'
+import TransactionRow from '@/features/transactionsDisplay/transactionRow/TransactionRow'
+import { useAppDispatch } from '@/hooks/redux'
+import { onEnterOrSpace } from '@/utils/misc'
+
+interface AddressTransactionListProps {
+  addressHash: AddressHash
+}
+
+const AddressTransactionsList = ({ addressHash }: AddressTransactionListProps) => {
```
Comment (mvaivre) [inline apps/desktop-wallet/src/features/transactionsDisplay/transactionLists/lists/AddressTransactionsList.tsx:38]: I really like how compact this is! 👏

File: apps/desktop-wallet/src/pages/UnlockedWallet/UnlockedWalletLayout.tsx (hunk 12 above)
Comment (mvaivre) [reply apps/desktop-wallet/src/pages/UnlockedWallet/UnlockedWalletLayout.tsx:171]: I guess we don't!

File: apps/explorer/src/pages/AddressInfoPage/AddressTransactionRow.tsx (hunk 16 above)
Comment (mvaivre) [reply apps/explorer/src/pages/AddressInfoPage/AddressTransactionRow.tsx:57]: yep. I agree it's still a bit confusing.

File: packages/shared/src/transactions.ts [hunk 21]
```diff
@@ -139,3 +144,10 @@ export const extractTokenIds = (tokenIds: Asset['id'][], ios: Transaction['input
     })
   })
 }
+
+export const findTransactionReferenceAddress = (addresses: AddressHash[], tx: Transaction | PendingTransaction) =>
+  addresses.find((address) => isAddressPresentInInputsOutputs(address, tx))
+
```
Comment (mvaivre) [inline packages/shared/src/transactions.ts:150]: We should test this thoroughly. I'm not saying it doesn't work, but there were many edge cases in the past so I'm careful :) Like what happens for an internal transfer, or some weird contract operations.

**Review by mvaivre — COMMENTED** [review]
WOW. Awesome piece of work! ⭐

File: apps/desktop-wallet/src/api/apiDataHooks/address/useFetchAddressBalancesAlph.ts (hunk 2 above)
Comment (nop33) [reply apps/desktop-wallet/src/api/apiDataHooks/address/useFetchAddressBalancesAlph.ts:29]: > With the precedent name I knew that it returned a transaction.

In fact, it is not returning a transaction. It was returning an object:

```ts
{
   latestTx: Transaction | undefined
   previousTx: Transaction | undefined
}
```

I wanted to find a name that is more appropriate for its usage, not for what it returns. The "signal" that we have outdated data is the latest transaction hash. I'm happy to discuss a different naming.

File: apps/desktop-wallet/src/api/apiDataHooks/wallet/useFetchWalletInfiniteTransactions.ts [hunk 22]
```diff
@@ -0,0 +1,60 @@
...
+import { useFetchWalletLastTransaction } from '@/api/apiDataHooks/wallet/useFetchWalletLastTransactions'
+import { walletTransactionsInfiniteQuery } from '@/api/queries/transactionQueries'
+import { useAppSelector } from '@/hooks/redux'
+import { selectAllAddressHashes } from '@/storage/addresses/addressesSelectors'
+
+const useFetchWalletInfiniteTransactions = () => {
+  const networkId = useAppSelector((s) => s.network.settings.networkId)
+  const allAddressHashes = useAppSelector(selectAllAddressHashes) // TODO: Display warning if > 80 addresses
+
+  const [fetchedTransactionListAt, setRefreshedTransactionListAt] = useState(0)
```
[the remark this answers is not in the document]
Comment (nop33) [reply apps/desktop-wallet/src/api/apiDataHooks/wallet/useFetchWalletInfiniteTransactions.ts:31]: Ah, I first had it `refreshed` and I renamed only the first part to `fetched`. I'd rather have `fetched`, WDYT? It's more accurate.

File: apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx
Comment (nop33) [reply apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx:1]: Hm, I also thought about it, but I like it as a separate feature. Displaying the sent transactions is not a requirement for the send functionality. Maybe we can rename the feature to `sendTransactionsDisplay`

File: apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx [hunk 23]
```diff
@@ -0,0 +1,183 @@
...
+import SnackbarBox from '@/features/snackbar/SnackbarBox'
+import { useAppDispatch, useAppSelector } from '@/hooks/redux'
+import { SentTransaction } from '@/types/transactions'
+
+const SentTransactionSnackbarPopup = memo(({ txHash }: { txHash: string }) => {
+  const dispatch = useAppDispatch()
+  const sentTx = useAppSelector((s) => selectSentTransactionByHash(s, txHash))
+  const [hide, setHide] = useState(false)
+
+  const { data: tx } = useQuery(pendingTransactionQuery({ txHash, skip: !sentTx || sentTx.status === 'confirmed' }))
```
[the remark this answers is not in the document]
Comment (nop33) [reply apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx:45]: Hm, I would go case by case. Not all our data fetching code is in the apiDataHooks. Some of it is in the features, like the chart for example.

File: apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx (hunk 18 above)
[the remark this answers is not in the document]
Comment (nop33) [reply apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx:118]: Would you like to work on it, with your amazing animation skills? :D IT'd be nice if the loading animation can morph into a check!

File: packages/shared/src/transactions.ts (hunk 21 above)
Comment (nop33) [reply packages/shared/src/transactions.ts:150]: This is the same code that we used to have (see `getTransactionsOfAddress` above), but yes, I agree.

File: apps/desktop-wallet/src/api/apiDataHooks/wallet/useFetchWalletInfiniteTransactions.ts (hunk 22 above)
[the remark this answers is not in the document]
Comment (mvaivre) [reply apps/desktop-wallet/src/api/apiDataHooks/wallet/useFetchWalletInfiniteTransactions.ts:31]: Works for me!

File: apps/desktop-wallet/src/api/apiDataHooks/address/useFetchAddressSingleTokenBalances.ts [hunk 24]
```diff
@@ -41,29 +41,25 @@ const useFetchAddressSingleTokenBalances = ({
...
     addressSingleTokenBalancesQuery({
       addressHash,
       networkId,
       tokenId,
-      latestTxHash: txHashes?.latestTxHash,
-      previousTxHash: txHashes?.latestTxHash,
-      skip: isLoadingTxHashes || isALPH || skip
+      latestTxHash: updatesSignal?.latestTx?.hash,
+      previousTxHash: updatesSignal?.previousTx?.hash,
+      skip: isLoadingUpdatesSignal || isALPH || skip
```
[the remark this answers is not in the document]
Comment (nop33) [reply apps/desktop-wallet/src/api/apiDataHooks/address/useFetchAddressSingleTokenBalances.ts]: Pushed

File: apps/desktop-wallet/src/api/apiDataHooks/wallet/useFetchWalletInfiniteTransactions.ts (hunk 22 above)
[the remark this answers is not in the document]
Comment (nop33) [reply apps/desktop-wallet/src/api/apiDataHooks/wallet/useFetchWalletInfiniteTransactions.ts]: Fixed and pished

File: apps/desktop-wallet/src/api/queries/transactionQueries.ts [hunk 25]
```diff
@@ -17,29 +17,33 @@ along with the library. If not, see <http://www.gnu.org/licenses/>.
...
 import queryClient from '@/api/queryClient'
 
 const ADDRESS_TRANSACTIONS_QUERY_KEYS = ['address', 'transactions']
 
-interface AddressLatestTransactionHashQueryProps {
+interface AddressLatestTransactionQueryProps {
   addressHash: AddressHash
   networkId: number
   skip?: boolean
+  pause?: boolean
```
[the remark this answers is not in the document]
Comment (nop33) [reply apps/desktop-wallet/src/api/queries/transactionQueries.ts]: Nice catch, I removed the code that was using it. Fixed and pushed

File: apps/desktop-wallet/src/features/send/Transfer/index.tsx
Comment (nop33) [reply apps/desktop-wallet/src/features/send/Transfer/index.tsx:1]: I agree. I can rename in my last PR to avoid conflicts.

File: apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx (hunk 23 above)
[the remark this answers is not in the document]
Comment (mvaivre) [reply apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx:45]: Ok. I just feel like fetching a pending TX is a pretty generic call (vs. feature specific). 
I still like the idea of this "fetching hook library" because it avoids duplicating the use of queries in component. What I mean by that: you develop a new feature, you need some data (say, a pending TX data), you look into this hook library (super easy to find what you're looking for thanks to our naming scheme) instead of relying on a custom `useQuery`.

But it's as you wish!

[the remark this answers is not in the document]
Comment (nop33) [reply apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx:45]: I don't have a strong opinion, whether we do `useQuery(pendingTransactionQuery` or `useFetchPendingTx` is pretty much the same for me. We already have a library of queries where `pendingTransactionQuery` is defined. I try to avoid unnecessary boilerplate code if it's not needed.

We have 2 uses of this query currently:

```ts
const { data: tx } = useQuery(pendingTransactionQuery({ txHash, skip: !sentTx || sentTx.status === 'confirmed' }))
```

```ts
const { data: pendingTx } = useQuery(pendingTransactionQuery({ txHash, skip: !isPendingTx }))
```

If you want, I can create a `useFetchPendingTransaction` that accepts the same params as the `pendingTransactionQuery` and it's simply a wrapper around `useQuery`.

File: apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx (hunk 18 above)
[the remark this answers is not in the document]
Comment (nop33) [reply apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx:118]: I wanted to somehow visualize the progress of moving from "sent" to "mempooled" to "confirmed". So I went with `10%` progress increase for "sent", `25%` for "mempooled" with the message "your tx is about to be included in the blockchain" and then gradually increase by `1.5%` per second to a max of `90%` where we stop and wait before the tx comes "confirmed" and we move the progress to `100%`.

[the remark this answers is not in the document]
Comment (nop33) [reply apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx:118]: We could also do:
- `10%` sent
- `50%` mempooled
- `100%` confirmed

with no fake progress. But I thought since our blocktime is 16s, a fake progress would not be THAT fake in most of the cases.

File: apps/desktop-wallet/src/features/sentTransactions/sentTransactionsSlice.ts (hunk 19 above)
Comment (nop33) [reply apps/desktop-wallet/src/features/sentTransactions/sentTransactionsSlice.ts]: Done!

File: apps/desktop-wallet/src/api/apiDataHooks/wallet/useFetchWalletLastTransactions.ts [hunk 26]
```diff
@@ -0,0 +1,79 @@
...
+  ),
+  ...combineIsLoading(results)
+})
+
+const extractLastTransactionHashes = (results: UseQueryResult<AddressLatestTransactionQueryFnData>[]) => ({
+  data: results
+    .flatMap(({ data }) =>
+      data
+        ? { addressHash: data.addressHash, latestTxHash: data?.latestTx?.hash, previousTxHash: data?.previousTx?.hash }
+        : undefined
```
[the remark this answers is not in the document]
Comment (nop33) [reply apps/desktop-wallet/src/api/apiDataHooks/wallet/useFetchWalletLastTransactions.ts]: Fixed!

File: apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx (hunk 23 above)
[the remark this answers is not in the document]
Comment (mvaivre) [reply apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx:45]: I think I would like it! :)

File: apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx (hunk 18 above)
[the remark this answers is not in the document]
Comment (mvaivre) [reply apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx:118]: Ok! Let's keep the progress steps then. I'll redesign this using a circle progress bar to be more subtle ;)

File: apps/explorer/src/pages/AddressInfoPage/AddressTransactionRow.tsx (hunk 16 above)
Comment (mvaivre) [reply apps/explorer/src/pages/AddressInfoPage/AddressTransactionRow.tsx:57]: Should we add a TODO somewhere?

Comment (nop33) [reply apps/explorer/src/pages/AddressInfoPage/AddressTransactionRow.tsx:57]: Yeah, probably in the shared package. Added here: bab402e50

File: apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx (hunk 23 above)
[the remark this answers is not in the document]
Comment (nop33) [reply apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx:45]: ok, so you'd like an entire new file just for this :P 

image

Yes, my lord. It shall be done.

[the remark this answers is not in the document]
Comment (nop33) [reply apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx:45]: I thought we were gonna use Tanstack to reduce Redux boilerplate code and now we are adding our own haha

File: apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx
Comment (nop33) [reply apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx:1]: On second thought, it's not only about displaying. The slice serves as a repository of actions the user took (txs they sent) which then is used for cache invalidation. I wouldn't rename it. But I can move the whole folder inside the `send` feature.

File: apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx (hunk 23 above)
[the remark this answers is not in the document]
Comment (nop33) [reply apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx:45]: Done in f6cc222ed

File: apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx
Comment (nop33) [reply apps/desktop-wallet/src/features/sentTransactions/SentTransactionSnackbarPopup.tsx:1]: Done in 2d9de0715

**Review by mvaivre — APPROVED** [review]
